provider/pkg/gen/go-templates/configFile.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2016-2020, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// *** WARNING: this file was generated by pulumigen. ***
+// *** Do not edit by hand unless you're certain you know what you are doing! ***
+
+package yaml
+
+import (
+	"github.com/pkg/errors"
+	"github.com/pulumi/pulumi/sdk/v2/go/pulumi"
+)
+
+type ConfigFile struct {
+	pulumi.ResourceState
+
+	resources pulumi.MapOutput
+}
+
+type ConfigFileArgs struct {
+	// File is a path or URL that uniquely identifies a file.
+	// TODO(joe): the .NET client seems to accept an input. Isn't that wrong? It will create resources inside Apply!
+	File string
+	// Transformations is an optional list of transformations to apply to Kubernetes resource definitions
+	// before registering with the engine.
+	Transformations []Transformation
+	// ResourcePrefix isn optional prefix for the auto-generated resource names. For example, a resource named `bar`
+	// created with resource prefix of `"foo"` would produce a resource named `"foo-bar"`.
+	ResourcePrefix string
+}
+
+func NewConfigFile(ctx *pulumi.Context,
+	name string, args *ConfigFileArgs, opts ...pulumi.ResourceOption) (*ConfigFile, error) {
+
+	// Register the resulting resource state.
+	configFile := &ConfigFile{
+		resources: pulumi.MapOutput{},
+	}
+	err := ctx.RegisterComponentResource("kubernetes:yaml:ConfigFile", name, configFile, opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	// Now provision all child resources by parsing the YAML file.
+	if args != nil {
+		// Make the component the parent of all subsequent resources.
+		opts = append(opts, pulumi.Parent(configFile))
+
+		// Honor the resource name prefix if specified.
+		if args.ResourcePrefix != "" {
+			name = args.ResourcePrefix + "-" + name
+		}
+
+		// Parse and decode the YAML files.
+		rs, err := parseDecodeYamlFiles(ctx, &ConfigGroupArgs{
+			Files:           []string{args.File},
+			Transformations: args.Transformations,
+			ResourcePrefix:  args.ResourcePrefix,
+		}, true, opts...)
+		if err != nil {
+			return nil, err
+		}
+		if rs != nil {
+			configFile.resources = *rs
+		}
+
+		// Finally, register all of the resources found.
+		err = ctx.RegisterResourceOutputs(configFile, pulumi.Map{})
+		if err != nil {
+			return nil, errors.Wrapf(err, "registering child resources")
+		}
+	}
+
+	return configFile, nil
+}
+
+func (cf *ConfigFile) Resources() pulumi.MapOutput {
+	return cf.resources
+}
```

Comment:
I thought out standard pattern was just to expose public fields?  We do that in https://github.com/pulumi/pulumi-docker/blob/master/sdk/go/docker/image.go for example.  @pgavlin @EvanBoyle ?